### PR TITLE
global: fix tests, fix Celery, extend tests

### PR DESCRIPTION
**Pull request by jacquerie**
@lnielsen As promised, here's the PR.

* The first commit fixes a failure in the tests by sharing the application context between all tests.
* The second introduces the feature I was interested in: compatibility with Celery + MessagePack.
* The third does what you asked me to do: shows that there's limited support for attachments and dates.

### CODE REVIEW

File: tests/test_invenio_mail_tasks.py [hunk 1]
```diff
@@ -64,3 +70,46 @@ def test_send_message_stream(email_task_app):
...
+
+
+def test_send_message_with_attachments(email_task_app):
+    """Test sending a message with attachments."""
+    with email_task_app.app_context():
+        filename = pkg_resources.resource_filename(
+            __name__, os.path.join('attachments', 'invenio.svg'))
+        content_type = 'image/svg+xml'
+        data = pkg_resources.resource_string(
+            __name__, os.path.join('attachments', 'invenio.svg'))
```
Comment (jacquerie) [inline tests/test_invenio_mail_tasks.py]: Looks ugly. Maybe I can hide it in a fixture? Or I'd love to have a solution which doesn't read from the FS.

File: tests/test_invenio_mail_tasks.py [hunk 2]
```diff
@@ -64,3 +70,46 @@ def test_send_message_stream(email_task_app):
...
+
+
+def test_send_message_with_date(email_task_app):
+    """Test sending a message with a date."""
+    with email_task_app.app_context():
+        msg = {
+            'subject': 'Test4',
+            'sender': 'test4@test4.test4',
+            'recipients': ['test4@test4.test4'],
+            'date': 1456242014.398119
```
Comment (lnielsen) [inline tests/test_invenio_mail_tasks.py]: Shouldn't this be ``datetime``?

File: tests/test_invenio_mail_tasks.py [hunk 3]
```diff
@@ -64,3 +70,46 @@ def test_send_message_stream(email_task_app):
...
+
+        attachments = [Attachment(filename, content_type, data)]
+        msg = {
+            'subject': 'Test3',
+            'sender': 'test3@test3.test3',
+            'recipients': ['test3@test3.test3'],
+            'attachments': attachments
+        }
+
+        send_email(msg)
```
Comment (lnielsen) [inline tests/test_invenio_mail_tasks.py]: Here the task function is called directly (vs ``send_email.delay()``). Have you tested if it works when you execute it async (i.e. ``CELERY_EAGER=False`` + msgpack serialiser + running worker)?

**Comment by lnielsen** [comment]
@jacquerie Thanks a lot for fixing it. The change to ``send_email`` looks good to me. However, I'm surprised if the test with attachments works async (and, if it works then you could likely send a message through e.g. rabbitmq on a couple of MBs).

I think the PR is probably good as is, but we should add an issue to add async tests (e.g. Invenio-Indexer has some tests with RabbitMQ - https://github.com/inveniosoftware/invenio-indexer/blob/master/.travis.yml#L51)

File: tests/test_invenio_mail_tasks.py (hunk 2 above)
Comment (jacquerie) [inline tests/test_invenio_mail_tasks.py]: You mean the key, or the type value? The key string should be date, according to https://pythonhosted.org/Flask-Mail/#flask_mail.Message.

As for the type, I lazily put a float there (the output of `import time; time.time()`).

File: tests/test_invenio_mail_tasks.py (hunk 3 above)
Comment (jacquerie) [inline tests/test_invenio_mail_tasks.py]: Whoops. Stupid mistake. Good news is that it works with `delay` and `CELERY_ALWAYS_EAGER=True`. Let me test with everything set up.

File: tests/test_invenio_mail_tasks.py (hunk 2 above)
Comment (lnielsen) [inline tests/test_invenio_mail_tasks.py]: It was the type i meant. I can see you're correct in using time:
https://github.com/mattupstate/flask-mail/blob/master/flask_mail.py#L184-L185

**Comment by lnielsen** [comment]
Also, please amend copyright years in headers.

**Comment by jacquerie** [comment]
> Also, please amend copyright years in headers.

:dancer:

File: tests/test_invenio_mail_tasks.py (hunk 2 above)
Comment (jacquerie) [inline tests/test_invenio_mail_tasks.py]: Yeah, that's what I saw as well.

**Comment by jacquerie** [comment]
Comments addressed and green. It remains to be seen if RabbitMQ + MessagePack + Attachments works, but if I understood it right it's a matter for another issue/PR.

**Pull request merged**